File: SyntaxGraphicsCompilerLib/LabelHandler.cpp
#include "StdAfx.h"
#include "LabelHandler.h"
#include "LabelCompiler.h"

using namespace Syntax::Compiler;
using namespace Syntax::GraphicsCompiler;

LabelHandler::LabelHandler(void)
{
}

LabelHandler::~LabelHandler(void)
{
}

void LabelHandler::HandleAttribute(const std::wstring &name,const std::wstring& value)
{
	if(name == L"name")
		compiler->Name(value);
	else if(name == L"font")
		compiler->Font(value);
	else if(name == L"text")
		compiler->Text(value);
	else if(name == L"anchorleft" && value == L"true")
		compiler->Anchor(0x01);
	else if(name == L"anchorright" && value == L"true")
		compiler->Anchor(0x02);
	else if(name == L"anchortop" && value == L"true")
		compiler->Anchor(0x04);
	else if(name == L"anchorbottom" && value == L"true")
		compiler->Anchor(0x08);
}

Syntax::Compiler::INodeCompiler* LabelHandler::StartElement()
{
	compiler = new LabelCompiler();
	compiler->Anchor(0);
	return compiler;
}

File: SyntaxGraphicsXML/Handlers/LabelHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Syntax.XMLParserLibrary;
using System.Xml.Linq;

namespace Syntax.Graphics.Handlers
{
    internal class LabelHandler : Handler<Label>
    {
        public override XElement Document
        {
            set { labels = value.Element("labels"); }
        }

        public override string XMLItemTag
        {
            get { return "label"; }
        }

        public override XElement CollectionElement
        {
            get { return labels; }
        }

        public override XMLData BuildData(XElement element)
        {
            Label result = new Label();
            result.Name = element.Attribute("name").Value;
            result.Font = element.Attribute("font").Value;
            result.Text = element.Attribute("text").Value;
            result.AnchorLeft = Boolean.Parse(element.Attribute("anchorleft").Value);
            result.AnchorRight = Boolean.Parse(element.Attribute("anchorright").Value);
            result.AnchorTop = Boolean.Parse(element.Attribute("anchortop").Value);
            result.AnchorBottom = Boolean.Parse(element.Attribute("anchorbottom").Value);
            return result;
        }

        public override void UpdateElement(XElement element, XMLData data)
        {
            Label label = data as Label;
            element.SetAttributeValue("name", label.Name);
            element.SetAttributeValue("font", label.Font);
            element.SetAttributeValue("text", label.Text);
            element.SetAttributeValue("anchorleft", label.AnchorLeft);
            element.SetAttributeValue("anchorright", label.AnchorRight);
            element.SetAttributeValue("anchortop", label.AnchorTop);
            element.SetAttributeValue("anchorbottom", label.AnchorBottom);
        }

        private XElement labels;
    }
}
